src/Integration.Vsix/SonarLintDaemon/DaemonIssueAdapter.cs

```diff
@@ -0,0 +1,55 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    internal class DaemonIssueAdapter : IFilterableIssue
+    {
+        public DaemonIssueAdapter(Sonarlint.Issue sonarLintIssue, string wholeLineText, string lineHash)
+        {
+            SonarLintIssue = sonarLintIssue ?? throw new ArgumentNullException(nameof(sonarLintIssue));
+
+            WholeLineText = wholeLineText;
+            LineHash = lineHash;
+        }
+
+        public Sonarlint.Issue SonarLintIssue { get; }
+
+        #region ILiveIssue interface
```

Comment:
In general, I'm not a big fan of regions, I'm used to seeing it in badly written code.
I think it can be removed here, the class doesn't do much besides implementing the interface.

Reply:
It's here to make it clearer what isn't part of the interface. Explicitly implementing the interface would be an alternative, but I find the syntax for that ugly and harder to read.

Reply:
Oh, I didn't understand that at all. I think implementing the interface would be better.
Why do you think it's ugly? We can have a new interface for both this class and LiveIssue.